packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -66,33 +70,88 @@ class _TextSelectionToolbarState extends State<_TextSelectionToolbar> with Ticke
     );
   }
 
+  // Asynchronously check if the clipboard has data and update
+  // _isClipboardEmpty if needed.
+  void _updateClipboardStatus() {
```

Comment:
nit: should this check if `widget.handlePaste` is not null first, in case the information isn't needed at all?

Reply:
+1 (see my comment above)